### PR TITLE
Update `HttpStatus` to `HttpStatusCode` in `ResponseEntityExceptionHandler` implementations

### PR DESCRIPTION
**Pull request by Laurens-W**
## What's your motivation?

- Fixes https://github.com/openrewrite/rewrite-spring/issues/369

## Anyone you would like to review specifically?

@timtebeek 

## Any additional context

- https://github.com/spring-projects/spring-framework/issues/28214
- https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpStatus.html
- https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpStatusCode.html

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
+                  }
+              }
+              """,
+            """
+              import org.springframework.http.HttpHeaders;
+              import org.springframework.http.HttpStatus;
```
Comment (Laurens-W) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:57]: In this particular test i'd expect this import to be removed, but the LST still seems to have a reference somewhere

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+              import org.springframework.http.HttpStatusCode;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatusCode status, WebRequest request) {
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
```
Comment (timtebeek) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:66]: The import is not removed due to a dangling type reference to `HttpStatus` on parameters going into this `super` method call.

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      HttpStatus enumValue = status.value();
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
+                  }
+              }
+              """,
+            """
+              import org.springframework.http.HttpHeaders;
+              import org.springframework.http.HttpStatusCode;
```
Comment (timtebeek) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:138]:
```suggestion
              import org.springframework.http.HttpStatus;
              import org.springframework.http.HttpStatusCode;
```

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+              import org.springframework.http.HttpStatus;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      int value = status.value();
```
Comment (timtebeek) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:91]: Over here `status.value()` returns an `int`...

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+              import org.springframework.http.HttpStatus;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      HttpStatus enumValue = status.value();
```
Comment (timtebeek) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:132]: ... whereas over here it returns an `HttpStatus`. That reads kind of weird to me; what's the difference between these two text blocks?

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode())
+          .expectedCyclesThatMakeChanges(2)
```
Comment (timtebeek) [inline src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java:31]: This is usually an indication that we're not making changes in the right order, or matching twice on the same elements. If you can't work it out then I'll need to dive in for a fix.

File: src/main/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode.java [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    private static final MethodMatcher HANDLER_METHOD = new MethodMatcher(
+            "org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler *(..)", true);
+
+    private static final MethodMatcher HTTP_STATUS_VALUE_OF = new MethodMatcher(
+            HTTP_STATUS_FQ + " valueOf(int)", true);
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesMethod<>(HANDLER_METHOD),
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode.java:58]: Can we add a test that does not call `super`? I wonder if we match those cases where we merely define a method, not use it.

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java (hunk 5 above)
Comment (Laurens-W) [reply src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java]: This specific usage was invalid, I've changed the testcase to use HttpStatus enum as that implements HttpStatusCode and therefore is a valid type to return.

File: src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java (hunk 6 above)
Comment (Laurens-W) [reply src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java]: Recipe now completes in 1 cycle :)

File: src/main/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode.java (hunk 7 above)
Comment (Laurens-W) [reply src/main/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode.java]: Added a test case that doesn't call super, which initially didn't match, so adjusted that

**Review by timtebeek — APPROVED** [review]
Good to see this covered in rewrite-spring; thanks for picking it up!

**Comment by timtebeek** [comment]
Failure also seen on main; let's pick that up separately. Thanks again!

**Pull request merged**